tests/tests_sign_workflow/test_sign_artifacts.py

```diff
@@ -0,0 +1,40 @@
+import unittest
+from pathlib import Path
+
+from sign_workflow.sign_artifacts import SignArtifacts, SignArtifactsExistingArtifactFile, SignExistingArtifactsDir, SignWithBuildManifest
+
+
+class TestSignArtifacts(unittest.TestCase):
+
+    def test_signer_class_method(self):
+        path = Path(r"/dummy/path/manifest.yml")
+        klass = SignArtifacts.__signer_class__(path)
+
+        self.assertEqual(type(SignWithBuildManifest), type(klass))
+
+        path = Path(r"/dummy/path/")
+        klass = SignArtifacts.__signer_class__(path)
+
+        self.assertEqual(type(SignExistingArtifactsDir), type(klass))
+
+        path = Path(r"/dummy/path/artifact.tar.gz")
+        klass = SignArtifacts.__signer_class__(path)
+
+        self.assertEqual(type(SignArtifactsExistingArtifactFile), type(klass))
+
+    def test_from_path_method(self):
```

Comment:
This ends up instantiating an instance of `Signer` and asks for git credentials. You have three options:

- patch `Signer` so it becomes a mock, this is preferred 
- make it a noop with `@patch("sign_workflow.signer.GitRepository")`, this is not recommended because it implies knowledge of what `Signer` is doing, one level too deep
- remove that instantiating from the constructor of `SignArtifacts`, which is not recommended because it will be executing `git checkout` every time we sign anything